### PR TITLE
CSP-1123 Implements PUT MemberProfile Endpoint in Outer API

### PR DESCRIPTION
**Pull request by ManishAgarwalDfE**

### CODE REVIEW

File: src/ApprenticeAan/SFA.DAS.ApprenticeAan.Api.UnitTests/Controllers/MyApprenticeshipControllerTests/UpdateMemberProfileAndPreferencesTests.cs [hunk 1]
```diff
@@ -0,0 +1,41 @@
+﻿
```
Comment (MarkFCain) [inline src/ApprenticeAan/SFA.DAS.ApprenticeAan.Api.UnitTests/Controllers/MyApprenticeshipControllerTests/UpdateMemberProfileAndPreferencesTests.cs:1]: remove this file, or uncomment, whichever is appropriate

**Review by MarkFCain — APPROVED** [review]

**Review by MarkFCain — APPROVED** [review]

**Review by MarkFCain — APPROVED** [review]

**Review by MarkFCain — APPROVED** [review]

File: src/ApprenticeAan/SFA.DAS.ApprenticeAan.Application/Members/Queries/UpdateMember/UpdateMemberCommand.cs
Comment (DevenKShah) [inline src/ApprenticeAan/SFA.DAS.ApprenticeAan.Application/Members/Queries/UpdateMember/UpdateMemberCommand.cs:1]: Delete if not used. Or put under correct folder

File: Directory.Packages.props [hunk 2]
```diff
@@ -23,7 +23,7 @@
     <PackageVersion Include="Microsoft.AspNetCore.Diagnostics.HealthChecks" Version="2.2.0" />
     <PackageVersion Include="Microsoft.AspNetCore.Hosting" Version="2.2.7" />
     <PackageVersion Include="Microsoft.AspNetCore.Hosting.Abstractions" Version="2.2.0" />
-    <PackageVersion Include="Microsoft.AspNetCore.JsonPatch" Version="8.0.0-preview.2.23153.2" />
+    <PackageVersion Include="Microsoft.AspNetCore.JsonPatch" Version="6.0.21" />
```
Comment (DevenKShah) [inline Directory.Packages.props:26]: Update to proper 8.0 version here and also in the inner api

File: src/ApprenticeAan/SFA.DAS.ApprenticeAan.Application/Members/Queries/UpdateMember/UpdateMemberCommand.cs
Comment (ManishAgarwalDfE) [reply src/ApprenticeAan/SFA.DAS.ApprenticeAan.Application/Members/Queries/UpdateMember/UpdateMemberCommand.cs:1]: Done

File: Directory.Packages.props (hunk 2 above)
Comment (ManishAgarwalDfE) [reply Directory.Packages.props]: Done

File: src/ApprenticeAan/SFA.DAS.ApprenticeAan.Api/Controllers/MembersController.cs [hunk 3]
```diff
@@ -22,4 +27,31 @@ public async Task<IActionResult> GetMembers([FromQuery] GetMembersRequestModel r
...
+        CancellationToken cancellationToken)
+    {
+        PatchMemberRequest patchMemberRequest = request.patchMemberRequest;
+
+        JsonPatchDocument<PatchMemberRequest> jsonPatchDocument = new JsonPatchDocument<PatchMemberRequest>();
+        if (patchMemberRequest.RegionId > 0)
+        {
+            jsonPatchDocument.Replace(x => x.RegionId, patchMemberRequest.RegionId);
+        }
+        jsonPatchDocument.Replace(x => x.OrganisationName, patchMemberRequest.OrganisationName);
```
Comment (DevenKShah) [inline src/ApprenticeAan/SFA.DAS.ApprenticeAan.Api/Controllers/MembersController.cs:46]: Why there is no check for organisation name, this should only be added if it has a value isn't it?

File: src/ApprenticeAan/SFA.DAS.ApprenticeAan.Api/Controllers/MembersController.cs [hunk 4]
```diff
@@ -22,4 +27,31 @@ public async Task<IActionResult> GetMembers([FromQuery] GetMembersRequestModel r
...
+        jsonPatchDocument.ApplyTo(patchMemberRequest);
```
[the remark this answers is not in the document]
Comment (ManishAgarwalDfE) [reply src/ApprenticeAan/SFA.DAS.ApprenticeAan.Api/Controllers/MembersController.cs:47]: Done

File: src/ApprenticeAan/SFA.DAS.ApprenticeAan.Api/Controllers/MembersController.cs (hunk 3 above)
Comment (ManishAgarwalDfE) [reply src/ApprenticeAan/SFA.DAS.ApprenticeAan.Api/Controllers/MembersController.cs:46]: Organisation name is nullable string so we shouldn't check for it if there is null then it is also valid value for it

Comment (DevenKShah) [reply src/ApprenticeAan/SFA.DAS.ApprenticeAan.Api/Controllers/MembersController.cs:46]: Not for Apprentices, this is a required information for Apprentices, see the onboarding journey.

File: src/ApprenticeAan/SFA.DAS.ApprenticeAan.Api/Controllers/MembersController.cs (hunk 4 above)
[the remark this answers is not in the document]
Comment (DevenKShah) [reply src/ApprenticeAan/SFA.DAS.ApprenticeAan.Api/Controllers/MembersController.cs:47]: The coverage is not satisfactory, you need to have separate tests for each piece of logic with specific assertions.

[the remark this answers is not in the document]
Comment (ManishAgarwalDfE) [reply src/ApprenticeAan/SFA.DAS.ApprenticeAan.Api/Controllers/MembersController.cs]: Done

File: src/ApprenticeAan/SFA.DAS.ApprenticeAan.Api/Controllers/MembersController.cs (hunk 3 above)
Comment (ManishAgarwalDfE) [reply src/ApprenticeAan/SFA.DAS.ApprenticeAan.Api/Controllers/MembersController.cs]: Done

**Review by DevenKShah — APPROVED** [review]

**Pull request merged**